pkg/apiserver/handlers.go

```diff
@@ -549,7 +553,7 @@ func (r *RequestInfoResolver) GetRequestInfo(req *http.Request) (RequestInfo, er
 
 			// if there is another step after the namespace name and it is not a known namespace subresource
 			// move currentParts to include it as a resource in its own right
-			if len(currentParts) > 2 {
+			if len(currentParts) > 2 && !namespaceSubresources.Has(currentParts[2]) {
```

Comment:
In order to be a subresource reference, it must not have another url segment.  If there's another segment, it seems more likely to an attempt on a namespaced resource and I'd expect our API server to route it accordingly.